### PR TITLE
Turn doc references into links

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

**Comment by charliermarsh** [comment]
\cc @not-my-profile - I assume this is good to do?

**Comment by not-my-profile** [comment]
Yes this is indeed good to do :) However I think we should firstly make the CI check such links if they can actually be resolved ... working on that right now :)

File: src/rules/pyupgrade/rules/remove_six_compat.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use crate::registry::{Diagnostic, RuleCode};
 use crate::source_code::{Generator, Locator, Stylist};
 use crate::violations;
 
-/// Return `true` if the `Expr` is a reference to `${module}.${any}`.
+/// Return `true` if the [`Expr`] is a reference to `${module}.${any}`.
 fn is_module_member(call_path: &[&str], module: &str) -> bool {
```
Comment (not-my-profile) [inline src/rules/pyupgrade/rules/remove_six_compat.rs:12]: nit: Which `Expr`? (I think this doc comment is outdated)

Comment (charliermarsh) [reply src/rules/pyupgrade/rules/remove_six_compat.rs:12]: Ah yeah. Definitely outdated.

**Pull request merged**